acceptance/builder_test.go

```diff
@@ -0,0 +1,367 @@
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+		when("CNB_BUILDPACKS_DIR changed", func() {
+			it("CNB_BUILDPACKS_DIR is successfully transmitted to in build script", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					"--env", "CNB_BUILDPACKS_DIR=/env_folders/different_buildpack_dir_from_env",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				//print(string(output), err)
+				h.AssertNil(t, err) //due to not exist directory
+				expected := "CNB_BUILDPACK_DIR: /env_folders/different_buildpack_dir_from_env"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+		when("CNB_LAYERS_DIR", func() {
+			it("CNB_LAYERS_DIR is successfully transmitted to in build script", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					"--env", "CNB_LAYERS_DIR=/tmp/different_layers_path_dir_from_env",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				//print(string(output), err)
+				h.AssertNil(t, err) //due to not exist directory
+				expected := "layers_dir: /tmp/different_layers_path_dir_from_env"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+		when("CNB_PLAN_PATH", func() {
+			it("CNB_PLAN_PATH is successfully transmitted to in build script", func() {
+
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/differrent_plan_from_env.toml",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				//print(string(output), err)
+				h.AssertNil(t, err) //due to not exist directory
+				expected := "different_plan_from_env.toml_reqires_subset_content"
+				h.AssertStringContains(t, string(output), expected)
+			})
+		})
+
+		when("CNB_PLATFORM_DIR", func() {
+			it("CNB_PLATFORM_DIR is successfully transmitted to in build script", func() {
+				command := exec.Command(
+					"docker",
+					"run",
+					"--rm",
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+					"--env", "CNB_PLATFORM_DIR=/different_platform_dir_from_env",
+					builderImage,
+				)
+				output, err := command.CombinedOutput()
+				//print(string(output), err)
+				h.AssertNotNil(t, err) //due to not exist directory
```

Comment:
I don't think this actually shows that `CNB_PLATFORM_DIR` made it all the way to the build script, as we'd be erroring out before any buildpacks are run. 

We could instead pass a real platform directory, and assert that the env was set correctly. Some context - if there's a /platform/env/MY_VAR file with contents MY_VAL, then buildpacks will have `MY_VAR=MY_VAL` in their environment.

Reply:
yeah i noticed this one and fixed. Thanks 👍